File: Low Level Design (LLD)/Problems/04. AirlineManagementSystem/Java/user/Passenger.java
package user;

import utils.UserRole;
import booking.Booking;

import java.util.ArrayList;
import java.util.List;

public class Passenger extends User {
    private List<Booking> bookings;

    public Passenger(String id, String name, String email, String phone) {
        this.id = id;
        this.name = name;
        this.email = email;
        this.phone = phone;
        this.role = UserRole.PASSENGER;
        this.bookings = new ArrayList<>();
    }

    public void addBooking(Booking booking) {
        bookings.add(booking);
    }

    public List<Booking> getBookings() {
        return bookings;
    }

    public String getEmail() {
        return email;
    }
}

File: Low Level Design (LLD)/Problems/04. AirlineManagementSystem/Cpp/user/src/Passenger.cpp
#include "../include/Passenger.h"

Passenger::Passenger(const std::string& id, const std::string& name, 
                     const std::string& email, const std::string& phone) :
              User(id, name, email, phone, UserRole::PASSENGER) {}

void Passenger::addBooking(Booking* booking) {
    bookings.push_back(booking);
}

const std::vector<Booking*>& Passenger::getBookings() const {
    return bookings;
}

std::string Passenger::getEmail() const {
    return email;
}